docs/release.md

```diff
@@ -0,0 +1,140 @@
+       checkout release-0.42.0
+       commit id: "Prepare release v0.42.0"
+       checkout main
+       merge release-0.42.0 id: "Pull Request"
+   ```
+
+   When testing and editing is completed, the sequence goes on.
+
+4. Maintainer merges the PR. GitHub removes the merged branch `release-0.42.0`.
+5. CI workflow triggers on merge:
+   1. Publishes the release page (`draft` → `latest`).
+   2. Moves the tag `v0.42.0` to the newly created merge commit by force-pushing a tag to GitHub.
+6. Another CI workflow triggers on the new tag push, checks that release is already published, and stops.
+7. Maintainer can now announce release to community.
+
+```mermaid
+gitGraph
+    commit id: "feature"
+    commit id: "feature 2"
+    commit id: "feature 3"
+    branch release-0.42.0
+    checkout release-0.42.0
+    commit id: "Prepare release v0.42.0"
+    checkout main
+    merge release-0.42.0 id: "Release v0.42.0" tag: "v0.42.0"
+```
+
+## Patch Releases
+
+Making a patch release has a lot in common with regular release, with a couple of differences:
+
+* A release branch is used instead of `main`
+* Patch commits are cherry-picked to the release branch.
+* A pull request is opened against the release branch.
+
+
+Let's assume that we've released `v0.42.0` and development goes on.
+We have introduced a couple of new features and some fixes to features which we have released 
+in `v0.42.0`.
+
+Once problems were found and fixed, a patch release is due.
+
+```mermaid
+gitGraph
+   commit id: "Release v0.42.0" tag: "v0.42.0"
+    checkout main
+    commit id: "feature 4"
+    commit id: "patch 1"
+    commit id: "feature 5"
+    commit id: "patch 2"
+```
+
+
+1. Maintainer makes a release branch `release-0.42` and cherry-picks patch commits from `main` to `release-0.42`.
+   These must be only patches to features that were present in version `v0.42.0`.
+
+   Cherry-picking can be done as soon as each patch is merged in `main`,
+   or directly before the release.
+
+   ```mermaid
+   gitGraph
+       commit id: "Release v0.42.0" tag: "v0.42.0"
+       branch release-0.42
+       checkout main
+       commit id: "feature 4"
+       commit id: "patch 1"
+       commit id: "feature 5"
+       commit id: "patch 2"
+       checkout release-0.42
+       cherry-pick id: "patch 1"
+       cherry-pick id: "patch 2"
+   ```
+
+   When all relevant patch commits are cherry-picked, the branch is ready for releasing.
+
+2. Maintainer tags the `HEAD` commit of branch `release-0.42` as `v0.42.1`, then pushes it to GitHub.
+3. CI workflow triggers on tag push:
+    1. Creates a draft page for release `v0.42.1`, if it wasn't created before.
+    2. Takes code from tag `v0.42.1`, builds images, and pushes them to ghcr.io.
+    3. Makes a new commit with updated digests, pushes it to new branch `release-0.42.1`, and opens a PR to `release-0.42`.
+    4. Builds Cozystack distributives (except images) from this commit, uploads them to the release draft page.
```

Comment:
@kvaps does CI build Cozystack binaries from this newly created "Prepare Release x.y.z" commit, or from the originally tagged commit?

Reply:
CI triggered on specific commit, then it builds the release, then it commit result assets with `Prepare Release x.y.z` message.
After it get merged it moves tag to merged commit with assets.